### PR TITLE
added risk policy support

### PR DESCRIPTION
**Pull request by bartes**

### CODE REVIEW

File: src/main/java/io/castle/client/model/RiskPolicyResult.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    private String name;
+
+    /**
+     * type of the risk policy
+     */
+    private String type;
+
+
+    public String getId() {
```
Comment (psyho) [inline src/main/java/io/castle/client/model/RiskPolicyResult.java:34]: We should probably consider using lombok to get rid of all the getters and setters from the code, but that's for another time.

File: src/main/java/io/castle/client/model/RiskPolicyResult.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * name of the risk policy
+     */
+    private String name;
+
+    /**
+     * type of the risk policy
+     */
+    private String type;
```
Comment (psyho) [inline src/main/java/io/castle/client/model/RiskPolicyResult.java:31]: I think this should be an enum, not a string: https://github.com/castle/castle-risk_policies/blob/master/lib/castle/risk_policies/models/risk_policy/type.rb

When deserializing, we should make sure not to break client code if we introduce a new type, so I'd opt for logging a WARN and returning null for unknown values.

File: src/main/java/io/castle/client/model/RiskPolicyType.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    public static RiskPolicyType fromType(String type) {
+        if (type == null) {
+            return null;
+        }
+        try {
+            return RiskPolicyType.valueOf(type);
+        } catch (IllegalArgumentException e) {
+            // no op, use string compare functions.
+        }
+        if (type.compareToIgnoreCase(BOT.name()) == 0) {
```
Comment (marysieek) [inline src/main/java/io/castle/client/model/RiskPolicyType.java:24]: why not: ` if(type.equalsIgnoreCase(BOT.name())`?

File: src/main/java/io/castle/client/model/RiskPolicyType.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    BOT, AUTHENTICATION;
+
+    /**
+     * Returns an RiskPolicyType from a string representing its name.
+     *
+     * @param type string representing the name of the RiskPolicyType, case-insensitive
+     * @return the enum value matching the name, or null if it does not match any enum
+     */
+    public static RiskPolicyType fromType(String type) {
+        if (type == null) {
```
Comment (marysieek) [inline src/main/java/io/castle/client/model/RiskPolicyType.java:16]: I'd rather go with simple for loop here:
```
        for (RiskPolicyType riskPolicy : RiskPolicyType.getEnumConstants()) {
            if (riskPolicy.name().equalsIgnoreCase(type)) {
                return riskPolicy;
            }
        }
        return null;
```

File: src/main/java/io/castle/client/model/AuthenticateAction.java [hunk 5]
```diff
@@ -18,22 +18,10 @@
...
-        }
-        if (action.compareToIgnoreCase(ALLOW.name()) == 0) {
-            return ALLOW;
-        }
-        if (action.compareToIgnoreCase(DENY.name()) == 0) {
-            return DENY;
-        }
-        if (action.compareToIgnoreCase(CHALLENGE.name()) == 0) {
-            return CHALLENGE;
+        for (AuthenticateAction kind : AuthenticateAction.class.getEnumConstants()) {
```
Comment (marysieek) [inline src/main/java/io/castle/client/model/AuthenticateAction.java:21]: I love this change ❤️

**Pull request merged**